### PR TITLE
PRIME-2360 Community Site and HA Site Snapshot 

### PR DESCRIPTION
**Pull request by bergomi02**

### CODE REVIEW

**Comment by neophyte57** [comment]
Hours of Operation is missing 
image

**Comment by neophyte57** [comment]
Ideally these would be identical in layout:

Overview:
image

versus Snapshot:
image

**Comment by bergomi02** [comment]
> Hours of Operation is missing image

this is fixed.

**Comment by bergomi02** [comment]
> Ideally these would be identical in layout:
> 
> Overview: image
> 
> versus Snapshot: image

It should be matching much better now. However, because the snapshot component is shared between Community Site and HA Site. Some fields are not matching exactly for one or other.

**Comment by neophyte57** [comment]
Snapshot looks a lot more consistent with Overview layout 👍.  That said, 
- can the email address come after the phone numbers (similar to Overview layout)
- can the phone numbers be pretty-formatted (like on Overview)
- the addresses of Contacts seem to be using the Site's Address rather than the SA's Address?  

image

**Comment by bergomi02** [comment]
> Snapshot looks a lot more consistent with Overview layout 👍. That said,
> 
> * can the email address come after the phone numbers (similar to Overview layout)
> * can the phone numbers be pretty-formatted (like on Overview)
> * the addresses of Contacts seem to be using the Site's Address rather than the SA's Address?
> 
> image

now the contact is using the same component as the one used in site overview page. They should be the same now.

**Comment by neophyte57** [comment]
Non-PCHP sites don't have Remote Users ... can the section be left out in case of DP and CP?

image

**Comment by bergomi02** [comment]
> Non-PCHP sites don't have Remote Users ... can the section be left out in case of DP and CP?
> 
> image

It is fixed.

**Comment by neophyte57** [comment]
Snapshot field name should match what is in Overview, i.e. Deferred Business Licence Reason
image

**Comment by neophyte57** [comment]
When there are no Remote Users, should say "None"
image

**Comment by bergomi02** [comment]
@neophyte57 
I believe the couple issues you mentioned are fixed. However, without my IDIR at the moment. I can't verify them myself. Please let me know if it is still an issue.

**Comment by neophyte57** [comment]
> @neophyte57 I believe the couple issues you mentioned are fixed. However, without my IDIR at the moment. I can't verify them myself. Please let me know if it is still an issue.

The blank Remote Users issue is still a problem ☹️

**Comment by bergomi02** [comment]
> > @neophyte57 I believe the couple issues you mentioned are fixed. However, without my IDIR at the moment. I can't verify them myself. Please let me know if it is still an issue.
> 
> The blank Remote Users issue is still a problem ☹️

it is addressed.

File: prime-dotnet-webapi/Controllers/SitesController.cs [hunk 1]
```diff
@@ -1380,5 +1384,41 @@ public async Task<ActionResult> GetDeviceProviderSite(string deviceProviderId)
...
             return Ok(site);
         }
+
+        // GET: api/sites/5/site-submissions
+        /// <summary>
+        /// Gets the site submissions.
+        /// </summary>
+        /// <param name="siteId"></param>
+        [HttpGet("{siteId}/site-submissions", Name = nameof(GetSiteSubmissions))]
+        [Authorize(Roles = Roles.PrimeAdministrant)]
```
Comment (neophyte57) [inline prime-dotnet-webapi/Controllers/SitesController.cs:1394]: I think better as `Roles.ViewSite`

File: prime-dotnet-webapi/Controllers/SitesController.cs [hunk 2]
```diff
@@ -1380,5 +1384,41 @@ public async Task<ActionResult> GetDeviceProviderSite(string deviceProviderId)
...
+
+
+        // GET: api/sites/5/site-submission/6
+        /// <summary>
+        /// Gets the site submission.
+        /// </summary>
+        /// <param name="siteId"></param>
+        /// <param name="siteSubmissionId"></param>
+        [HttpGet("{siteId}/site-submission/{siteSubmissionId}", Name = nameof(GetSiteSubmission))]
+        [Authorize(Roles = Roles.PrimeAdministrant)]
```
Comment (neophyte57) [inline prime-dotnet-webapi/Controllers/SitesController.cs:1413]: I think better as `Roles.ViewSite`

File: prime-dotnet-webapi/Controllers/SitesController.cs (hunk 1 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Controllers/SitesController.cs]: It is updated.

File: prime-dotnet-webapi/Controllers/SitesController.cs (hunk 2 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Controllers/SitesController.cs]: It is updated.

File: prime-dotnet-webapi/Migrations/20240702173818_AddSiteSubmission.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                table: "Agreement");
+
+            migrationBuilder.AlterColumn<string>(
+                name: "Discriminator",
+                table: "HealthAuthorityContact",
+                type: "character varying(55)",
+                maxLength: 55,
+                nullable: false,
+                oldClrType: typeof(string),
+                oldType: "text");
```
Comment (neophyte57) [inline prime-dotnet-webapi/Migrations/20240702173818_AddSiteSubmission.cs:26]: Why is this part of Migration?

Comment (bergomi02) [reply prime-dotnet-webapi/Migrations/20240702173818_AddSiteSubmission.cs]: Recreated the migration. It seems all unrelated changes are gone.

File: prime-angular-frontend/src/app/modules/adjudication/pages/community-site-submission-list-page/community-site-submission-list-page.component.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<app-site-registration-container [content]="content"
+                                 [hasActions]="hasActions">
+  <app-page-header>Site Information</app-page-header>
```
[the remark this answers is not in the document]
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/pages/community-site-submission-list-page/community-site-submission-list-page.component.html]: changed.

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/site-submission/site-submission.component.html [hunk 5]
```diff
@@ -0,0 +1,292 @@
+<h6 class="text-green text-italic">Submitted {{ submission?.submittedDate | formatDate }}</h6>
+<app-page-subheader>
+  <ng-container appPageSubheaderTitle><mat-icon class="mr-2">store</mat-icon>Site Information</ng-container>
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/site-submission/site-submission.component.html:3]: `Site Submission` more informative

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/site-submission-list/site-submission-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import moment from 'moment';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/site-submission-list/site-submission-list.component.ts:2]: `EventEmitter` and `Output` not used

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/site-submission/site-submission.component.html (hunk 5 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/site-submission/site-submission.component.html]: Updated.

File: prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/ha-site-submission-page/ha-site-submission-page.component.html [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<app-health-authority-site-container [content]="content"
+                                     [hasActions]="hasActions">
+  <app-page-header>Site Submission</app-page-header>
```
[the remark this answers is not in the document]
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/ha-site-submission-page/ha-site-submission-page.component.html]: Updated.

File: prime-angular-frontend/src/app/modules/adjudication/pages/community-site-submission-list-page/community-site-submission-list-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { SiteResource } from '@core/resources/site-resource.service';
+import { TechnicalSupportPageComponent } from '@registration/pages/technical-support-page/technical-support-page.component';
```
[the remark this answers is not in the document]
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/pages/community-site-submission-list-page/community-site-submission-list-page.component.ts]: removed.

**Pull request merged**